nova_plugin/server.py

```diff
@@ -409,6 +410,17 @@ def connect_floatingip(nova_client, fixed_ip, **kwargs):
     server = nova_client.servers.get(server_id)
     server.add_floating_ip(floating_ip_address, fixed_ip or None)
 
+    try:
+        floating_ip_object = \
+            nova_client.floating_ips.find(ip=floating_ip_address)
```

Comment:
i'm not sure at all this is the right way to do it, as you're querying floating ips via nova (i.e. deprecated method of doing it); should either use neutron, or query the server for its network addresses and look up the floating IP there.
then, if it's not there, rather than raising a nonrecoverableerror, perhaps we should run a retry on the operation, until it does take hold.